array_linked_list.py/arrays.py

```diff
@@ -2,63 +2,103 @@
 
 # Do not use any of the built in array functions for this exercise
 class array:
-    def __init__(self):
+    def __init__(self, capacity):
+
         # Your code here
-        pass
+        self.capacity = capacity
+        self.elements = [None] * capacity
+        self.count = 0
 
 
 # Double the size of the given array
-def resize_array():
+def resize_array(array):
     # Your code here
-    pass
+    new_capacity = array.capacity * 2
+    new_elements = [None] * new_capacity
+
+    for i in range(array.capacity):
+        new_elements[i] = array.elements[i]
+
+    array.elements = new_elements
+    array.capacity = new_capacity
 
 
 # Return an element of a given array at a given index
-def array_read():
+def array_read(array, index):
     # Throw an error if array is out of the current count
     # Your code here
-    pass
+    if index >= array.count:
+        print(f'Error! index ' + str(index) + f'out of range')
+        return None
+    return array.elements[index]
 
 
 # Insert an element in a given array at a given index
-def array_insert():
+def array_insert(array, value, index):
     # Throw an error if array is out of the current count
-
+    if index > array.count:
```

Comment:
The built-in insert method for lists in Python 3 allows you to insert items at an index greater than or equal to the length of the list. If the index is greater than or equal to the length of the list, the item will automatically be appended to the end of the list.

The example below will print out [1, 2, 3, 4] when executed in Python: 

```
l = [1, 2, 3]
l.insert(100, 4)
print(l)
```

Also, the built-in insert method for lists in Python 3 allows you to insert items at an index less than the negative conversion of the length of the list. If the index is less than the negative conversion of the length of the list, the item will automatically be added to the beginning of the list.

The example below will print out [0, 2, 3, 4] when executed in Python: 

```
l = [1, 2, 3]
l.insert(-700, 0)
print(l)
```

Reply:
I was not aware that insert did this. Great info on this Eddy thank you